sdk/schema/executable.go

```diff
@@ -119,79 +113,42 @@ func (e Executable) Command() string {
 
 func (c CredentialUsage) Validate() (bool, ValidationReport) {
 	report := ValidationReport{
-		Checks: []ValidationCheck{},
+		Heading: fmt.Sprintf("Credential usage %s", c.ID()),
+		Checks:  []ValidationCheck{},
 	}
-	credRef, err := c.GetCredentialReference()
 	report.AddCheck(ValidationCheck{
-		Description: "If defined, a credential reference must have at least a Name",
-		Assertion:   err == nil,
+		Description: "If defined, a credential reference must have at least a `Name`",
+		Assertion:   c.Name != "" || (c.Plugin == "" && c.Provisioner == nil),
 		Severity:    ValidationSeverityError,
 	})
 
-	selection, err := c.GetCredentialSelection()
 	report.AddCheck(ValidationCheck{
-		Description: "If defined, a credential selection must have its ID and IncludeAllCredentials set",
-		Assertion:   err == nil,
+		Description: "If defined, a credential selection must have its `ID` and `IncludeAllCredentials` set",
+		Assertion:   c.SelectFrom == nil || (c.SelectFrom.ID != "" && c.SelectFrom.IncludeAllCredentials),
 		Severity:    ValidationSeverityError,
 	})
 
-	id, err := c.ID()
 	report.AddCheck(ValidationCheck{
 		Description: "Credential usage has either a credential reference or selection properly defined, but not both",
-		Assertion:   err == nil && !(credRef != nil && selection != nil),
+		Assertion:   c.ID() != "" && !(c.SelectFrom != nil && c.Name != ""),
```

Comment:
Can we write this check without using `c.ID()`? That function depends on `Validate()`. By also using it here, we have a circular dependency: `ID()`'s output is valid if `Validate()` succeeds, which success depends on the output of `ID()`.

Reply:
Yes, makes sense! Done, thanks!